Sources/Brave/Frontend/Rewards/BraveRewards.swift

```diff
@@ -88,13 +85,18 @@ public class BraveRewards: NSObject {
         wallet.paymentId,
         base64Seed: Data(seed).base64EncodedString()
       )
+      self.ads.initialize() { success in
```

Comment:
Its possible this flow fails if `GetRewardsWallet` on the ledger side returns a `nullptr` for the wallet, therefore we have to change this flow a bit. If wallet is nil what is the flow? Do we initialize ads anyways? If not, then `isAdsInitialized` should be flipped back at the `guard let wallet = wallet else { return }` part